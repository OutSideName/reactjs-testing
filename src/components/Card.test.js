import {shallow, mount} from 'enzyme';
import React from 'react';
import Card from './Card';
import renderer from 'react-test-renderer';

it("testing render json", ()=>{
  const wrapper = renderer.create(<Card />).toJSON();
  expect(wrapper).toMatchSnapshot();
})

it("testing instance", ()=> {
  const wrapper = shallow(<Card />);
  expect(wrapper.instance()).toMatchSnapshot();
})



