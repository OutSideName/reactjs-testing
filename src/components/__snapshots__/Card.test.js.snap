// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing instance 1`] = `
Card {
  "context": Object {},
  "props": Object {},
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <Card />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
      >
        <img
          alt="robots"
          src="https://robohash.org/undefined?200x200"
        />
        <div>
          <h2 />
          <p />
        </div>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`testing render json 1`] = `
<div
  className="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
>
  <img
    alt="robots"
    src="https://robohash.org/undefined?200x200"
  />
  <div>
    <h2 />
    <p />
  </div>
</div>
`;
